support/node-health-check/src/main.rs

```diff
@@ -0,0 +1,113 @@
+use std::convert::Infallible;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::{sync::Arc, time::Duration};
+
+use anyhow::Result;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use structopt::StructOpt;
+use warp::Filter;
+
+/// Health struct returned by the RPC
+#[derive(Debug, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Health {
+    /// Number of connected peers
+    pub peers: usize,
+    /// Is the node syncing
+    pub is_syncing: bool,
+    /// Should this node have any peers
+    /// Might be false for local chains or when running without discovery.
+    pub should_have_peers: bool,
+}
+
+impl Health {
+    fn is_healthy(&self, required_peers: usize) -> bool {
+        if self.is_syncing {
+            return false;
+        }
+        if self.should_have_peers {
+            return self.peers >= required_peers;
+        }
+        true
+    }
+}
+
+#[derive(Debug, StructOpt, Clone)]
+#[structopt(
+    name = "node-health-check",
+    about = "Sidecar for substrate node to report health compatible with AWS ALB."
+)]
+struct Opt {
+    /// Debug
+    #[structopt(short, long)]
+    debug: bool,
+
+    /// Set poll_interval
+    #[structopt(short = "i", default_value = "5")]
+    poll_interval_in_secs: u64,
+
+    /// Set minimum peers to be connected for this node for it to be considered healthy
```

Comment:
Still a little confusing, maybe "The inspected node must have at least this many peers to be considered healthy".

Reply:
Agree, more succint, changed it